File: dashboard/src/main/java/de/dstoll/dashboard/service/MeetupEventScheduler.java
package de.dstoll.dashboard.service;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
@Slf4j
public class MeetupEventScheduler {

    private final EventDataReader eventDataReader;
    private final MeetupEventProducer meetupEventProducer;

    public void scheduleMeetupEvents(int numberOfEvents, long interval) {
        new Thread(() -> {
            try {
                for (int i = 0; i < numberOfEvents; i++) {
                    meetupEventProducer.sendEvent(eventDataReader.readData());
                    Thread.sleep(interval);
                }
            } catch (InterruptedException iex) {
                log.error("Kafka producer thread interrupted!");
            }
        }).start();
    }
}

File: kafka-producer/src/main/kotlin/de/dstoll/kafkaproducer/service/MeetupEventScheduler.kt
package de.dstoll.kafkaproducer.service

import org.springframework.stereotype.Service

@Service
class MeetupEventScheduler(
        private val eventDataReader: EventDataReader,
        private val meetupEventProducer: MeetupEventProducer
) {

    fun scheduleMeetupEvents(numberOfEvents: Int, interval: Long) {
        val dispatcher = Thread {
            repeat(numberOfEvents) {
                meetupEventProducer.sendEvent(eventDataReader.readData())
                Thread.sleep(interval)
            }
        }
        dispatcher.start()
    }
}